### DISCUSSION TITLE
Homebrew prompts that the formula does not exist when installing an existing formula

### DISCUSSION THREAD

**Discussion by TheColdVoid** [Homebrew/discussions#552] [Everyday usage]
When I tried to use homebrew to install Kafka,
`brew install kafka`
I got the following output:
` thevoid@TheVoidsMBP  ~  brew install kafka
==> Searching for similarly named formulae...
These similarly named formulae were found:
kafka                   kafkacat                librdkafka
To install one of them, run (for example):
  brew install kafka
Error: No available formula or cask with the name "kafka".
==> Searching taps on GitHub...
Error: No formulae found in taps.`
I can see that the "kafka" formula exists, but it says I can't install it. Can anyone help me with this problem?I'm not sure if this problem can be reproduced, so I'm asking here

**Comment by gromgit** [comment]
A common cause of that error is a damaged `kafka` formula. First try to reset the file contents from your local repo:
```
git -C $(brew --repo homebrew/core) checkout -- Formula/kafka.rb
```
then try installing it again. If that doesn't work, post the output of `brew audit --strict kafka`.
```

**Reply by iMichka** [reply]
I already saw that too. It's a little bit weird that you can't see the full stack trace when there is a typo or an error in a formula's content.

**Reply by Rylan12** [reply]
Running `brew readall` should show more traceback information

**Reply by TheColdVoid** [reply]
@Rylan12 @iMichka @gromgit 
`brew readall` output :https://gist.github.com/TheColdVoid/6d89e735975b065de8f15ef98efa2e8c

After running `git -C $(brew --repo homebrew/core) checkout -- Formula/kafka.rb`,I try to install kafka again,but that also doesn't work,got a same result:
`==> Searching for similarly named formulae...
These similarly named formulae were found:
kafka                   kafkacat                librdkafka
To install one of them, run (for example):
  brew install kafka
Error: No available formula or cask with the name "kafka".
==> Searching taps on GitHub...
Error: No formulae found in taps.`

I ran `brew audit --strict kafka` and got above output:
`Error: No available formula or cask with the name "kafka".`

Thank you all for helping me solve my problem

**Reply by carlocab** [reply]
What does `brew config` and `brew doctor` say?

**Reply by Rylan12** [reply]
Oh, I see. I misunderstood the issue a little bit.

It looks like you're just using an old version of Homebrew/core. Have you run `brew update`? See if that helps.

If not, you can try running `brew update-reset`. **Note: this will destroy all your uncommitted or committed changes in any Homebrew repository**

Otherwise, please post the output of `brew config` and `brew doctor`

**Reply by carlocab** [reply]
I suspect the core tap is old but `brew` is new. That's why it's reporting all these errors. I don't know how they got out of sync though.

**Reply by TheColdVoid** [reply]
I'm updating.

**Reply by gromgit** [reply]
@TheColdVoid , assuming you haven't made any changes to your local Homebrew repos, might be better to `brew update-reset` to force everything back into a sane config.

**Reply by TheColdVoid** [reply]
After the update this problem is solved.Thanks a  lot. @gromgit @carlocab @Rylan12

**Comment by xq9** [comment]
brew update-reset is useful for kafka software

**Reply by LufunoMulaudzi** [reply]
Thanks. This worked for me.
